###############
# TMSEG Method
###############

Readme by Muriel Keribin & Cyril Duchon-Doris

## Description

TMSEG predicts transmembrane proteins (TMP) and transmembrane helices (TMH) using position-specific scoring matrices (PSSM) generated by PSI-BLAST and the physico-chemical properties of the amino acids. The prediction is divided into three steps performed by three different classifiers. 

## Author

Michael Bernhofer - Student Summer 2013

bernhoferm.( at ).rostlab.org

First version of the program : work started in 2014, first commit on 09/02/2014, last one on 01/03/2014
A second version (TMSEG 2) was released by Michael in november 2015 on https://github.com/BernhoferM/TMSEG2. The repo was then forked to Rostlab/ namespace and work should continue from there

## Programming Language

Program installable with autotools for linux platforms.

The program itself is coded in java and exported in a .jar file. Source files available under src/

## Machine learning methods

 *  Random Forest (RF) decision trees
   *  predicts the probability of each residues to be in one of three states: transmembrane, soluble, and signal peptide. The RF uses a sliding window of 19 residues for the PSSM scores and 9 residues for the physico-chemical properties (charge, hydrophobicity, polarity). The protein sequence is then divided into transmembrane and soluble segments (and signal peptide, if applicable) based on the probabilities. 
   * predicts the inside/outside topology of the N-terminus. The prediction is based on the amino acid composition and positive charge of the residues on the two sides of the membrane (separated by the TMHs). 
 *  Neural Network (NN) refines the prediction by adjusting the position of the TMHs or potentially splitting very long TMHs (>36 residues). This NN is specifically trained on the length, amino acid composition, and physico-chemical properties of TMHs. 

 
## Inputs/Outputs

* IN -i : FASTA file (only amino acids sequences)
* IN -p : PSSM Matrix file generated by PSI-BLAST

* OUT -o : Human readable file
* OUT -r : Raw prediction scores

* FLAG -m : Multi-job (process whole folder of PSSM/FASTA)
* FLAG -x : Process previous prediction - Adjust (requires FASTA)
* FLAG -t : Only perform topology prediction

## How-to and compatibility

The program has been compiled into a .jar that can easily be executed :

``java -jar tmseg.jar -i examples/query.fasta -p examples/query.pssm -o tests/query.out``

The project can be loaded from eclipse (a .project and .classpath were added by us)

Because of JAVA portability, the program should work under any OS that can run a JVM v6+.
* WINDOWS : OK
* DEBIAN : OK
* UBUNTU : OK
* MacOSX : UNTESTED

## References :

* [1] L. Käll, A. Krogh, and E. L. Sonnhammer. An HMM posterior decoder for sequence feature prediction that includes homology information. Bioinformatics, 21 Suppl 1:i251–257, Jun 2005. [DOI:10.1093/bioinformatics/bti1014] [PubMed:15961464].
* [2] D. T. Jones. Improving the accuracy of transmembrane protein topology prediction using evolutionary information. Bioinformatics, 23(5):538–544, Mar 2007. [DOI:10.1093/bioinformatics/btl677] [PubMed:17237066].
* [3] B. Rost, P. Fariselli, and R. Casadio. Topology prediction for helical transmembrane proteins at 86% accuracy. Protein Sci., 5(8):1704–1718, Aug 1996. [PubMed Central:PMC2143485] [DOI:10.1002/pro.5560050824] [PubMed:8844859]. 

## Other notes :

Original wiki page : https://rostlab.org/owiki/index.php/Tmseg
Original repo (forked from) TMSEGv2 : https://github.com/BernhoferM/TMSEG2