###############
# TMSEG (TransMembrane SEGmentation)
###############
Stable version: `2.1.0`
See Wiki for more info

Muriel Keribin & Cyril Duchon-Doris for Protein Prediction II WinterSemester 2015-2016

TMSEG predicts transmembrane proteins (TMP) and transmembrane helices (TMH) using position-specific scoring matrices (PSSM) generated by PSI-BLAST and the physico-chemical properties of the amino acids.

## HOWTO Install

### This repo uses git-lfs

Some files are very large (model files for ML algorithms), and have been committed using git-lfs*. It is mandatory to install the git-lfs when cloning/pulling the repo. 
Download and install [git-lfs](https://git-lfs.github.com/)
Then run `git lfs install` on each machine once. Then you don't need to worry about it (only if you wish to add/rename some files that you want to track with git-lfs).

*Git Large File Storage (LFS) replaces large files such as audio samples, videos, datasets, and graphics with text pointers inside Git, while storing the file contents on a remote server like GitHub.com or GitHub Enterprise.

### Installation

The program is precompiled into a .jar, no other installation needed

## HOWTO RUN, Basics

Once pulled with git-lfs, you will find tmseg.jar in the first /src/ folder
This jar can easily be executed with java with some examples provided in the /example folder:

```shell
cd src/ 
java -jar tmseg.jar -i examples/query.fasta -p examples/query.pssm -o tests/query.out
cat tests/query.out
```

The project can be loaded from eclipse (a .project and .classpath were added by us)

Because of JAVA portability, the program works under any OS that can run a JVM. Java 1.7 and 1.8 are reported to work fine.

## Method Description

### Authors

Michael Bernhofer (1)
Edda Kloppmann (1,2)
Jonas Reeb (1)
Burkhard Rost (1,2,3,4)

1. Department of Informatics & Center for Bioinformatics & Computational Biology – i12, Technische Universität München (TUM), Boltzmannstr. 3, 85748 Garching/Munich, Germany
2. New York Consortium on Membrane Protein Structure, New York Structural Biology Center, 89 Convent Avenue, New York, NY 10027
3. Institute of Advanced Study (TUM-IAS), Lichtenbergstr. 2a, 85748 Garching/Munich, Germany
4. Institute for Food and Plant Sciences WZW – Weihenstephan, Alte Akademie 8, Freising, Germany

First version of the program : work started in 2014, first commit on 09/02/2014, last one on 01/03/2014
A second version (TMSEG 2) was released by Michael in november 2015 on https://github.com/BernhoferM/TMSEG2. The repo was then forked to Rostlab/ namespace and work should continue from there

Michael Bernhofer has implemented the method in this java program TMSEG

### Publications

The paper is not yet published (currently in progress).

### Training/Test Data

Will be detailed once the article is published

@Berhnhofer wrote:

```
I do have a dataset of transmembrane and soluble proteins, split into four equal parts. Split 1-3 are used for three-fold cross-training and cross-validation to optimize TMSEG's parameters. Split 4 is used at the end for the final (reported) performance.
Though v2.1.0's reported performance is only slithly different from v2.0.0, it is more robust when using different databases for the PSSM profiles.

I do have some java code for the training/optimization/validation, but not everything of it is completely automated. I can also give you the dataset and PSSM profiles if you want them, though I do not think they should be in the repository (over a thousand protein files and multiple PSSM profiles per protein).
```

### Programming Language

The program itself is coded in java and exported in a .jar file. Source files available under src/

### Machine learning methods

The prediction is divided into three steps performed by three different classifiers. 

 *  Random Forest (RF) decision trees
   *  predicts the probability of each residues to be in one of three states: transmembrane, soluble, and signal peptide. The RF uses a sliding window of 19 residues for the PSSM scores and 9 residues for the physico-chemical properties (charge, hydrophobicity, polarity). The protein sequence is then divided into transmembrane and soluble segments (and signal peptide, if applicable) based on the probabilities. 
   * predicts the inside/outside topology of the N-terminus. The prediction is based on the amino acid composition and positive charge of the residues on the two sides of the membrane (separated by the TMHs). 
 *  Neural Network (NN) refines the prediction by adjusting the position of the TMHs or potentially splitting very long TMHs (>36 residues). This NN is specifically trained on the length, amino acid composition, and physico-chemical properties of TMHs. 

## EVALUATION

### Performance

TMSEG was compared to three established methods: PolyPhobius [1], MEMSAT3 [2], and PHDhtm [3]. Its performance was at least comparable to the other three methods and often better (unpublished data). The evaluation was performed on two datasets with 41 and 60 transmembrane proteins, and 285 and 508 soluble proteins, respectively.

TMSEG correctly identified 98-100% of the transmembrane proteins (41 out of 41; 59 out of 60) and had a false positive rate of only 1-2% (6 out of 285; 5 out of 508). Transmembrane helices were predicted with a precision of 83-85% and recall of 84-85%, and 60-66% of all transmembrane proteins were predicted with all their helices at the correct positions (i.e. no false positives/negatives).

A predicted helix was considered to be correct if its end-points did not deviate by more than five residues from the observed helix, and if the overlap between the predicted and observed helix was at least half of the length of the longer helix.


### Limitations

TMSEG uses only the PSI-BLAST PSSM scores and features derived from those scores. Therefore, the quality of the prediction strongly depends on the quality of the PSSM. In order to approximate a lower bound, PSSMs from a PSI-BLAST run against an empty database were used (i.e. effectively using the BLOSUM62 matrix).

These poor PSSMs mainly affected the recall of transmembrane proteins and helices. The protein recall dropped to 83-90%, and the helix recall was around 52-56%. The precision of the transmembrane helices dropped to 70%, and the number of transmembrane proteins with all helices at their correct positions was only around 29-36%. However, the false positive rate (i.e. soluble proteins predicted as transmembrane proteins) was unaffected and remained at around 1%. 

## HOW-TO Extended
 
### Inputs/Output arguments and flags

* IN -i : FASTA file (only amino acids sequences)
* IN -p : PSSM Matrix file generated by PSI-BLAST

* OUT -o : Human readable file
* OUT -r : Raw prediction scores

* FLAG -m : Multi-job (process whole folder of PSSM/FASTA)
* FLAG -x : Process previous prediction - Adjust (requires FASTA)
* FLAG -t : Only perform topology prediction

## References :

* [1] L. Käll, A. Krogh, and E. L. Sonnhammer. An HMM posterior decoder for sequence feature prediction that includes homology information. Bioinformatics, 21 Suppl 1:i251–257, Jun 2005. [DOI:10.1093/bioinformatics/bti1014] [PubMed:15961464].
* [2] D. T. Jones. Improving the accuracy of transmembrane protein topology prediction using evolutionary information. Bioinformatics, 23(5):538–544, Mar 2007. [DOI:10.1093/bioinformatics/btl677] [PubMed:17237066].
* [3] B. Rost, P. Fariselli, and R. Casadio. Topology prediction for helical transmembrane proteins at 86% accuracy. Protein Sci., 5(8):1704–1718, Aug 1996. [PubMed Central:PMC2143485] [DOI:10.1002/pro.5560050824] [PubMed:8844859]. 

## Other notes :

Original wiki page : https://rostlab.org/owiki/index.php/Tmseg
Original repo (forked from) TMSEGv2 : https://github.com/BernhoferM/TMSEG2
