###############
# TMSEG Method
###############

Readme by Muriel Keribin & Cyril Duchon-Doris

## Description

MSEG predicts transmembrane proteins (TMP) and transmembrane helices (TMH) using position-specific scoring matrices (PSSM) generated by PSI-BLAST and the physico-chemical properties of the amino acids. The prediction is divided into three steps performed by three different classifiers. 

## Author

Michael Bernhofer - Student Summer 2013

bernhoferm.( at ).rostlab.org

Started work in 2004, first commit 09/02/2014, last on 01/02/2014
Released a second version (TMSEG 2) in november 2015

## Programming Language

Program installable with autotools (linux toolkit for dealing with portability)

The program itself is coded in java and exported in a .jar file. Not really clear if it can run on other OS than Linux though.

## Machine learning methods

 *  Random Forest (RF) decision trees
   *  predicts the probability of each residues to be in one of three states: transmembrane, soluble, and signal peptide. The RF uses a sliding window of 19 residues for the PSSM scores and 9 residues for the physico-chemical properties (charge, hydrophobicity, polarity). The protein sequence is then divided into transmembrane and soluble segments (and signal peptide, if applicable) based on the probabilities. 
   * predicts the inside/outside topology of the N-terminus. The prediction is based on the amino acid composition and positive charge of the residues on the two sides of the membrane (separated by the TMHs). 
 *  Neural Network (NN) refines the prediction by adjusting the position of the TMHs or potentially splitting very long TMHs (>36 residues). This NN is specifically trained on the length, amino acid composition, and physico-chemical properties of TMHs. 

 
## Inputs/Outputs

* IN : PSSM Matrix file
* IN : FASTA file (nucleotide sequence)
* OUT : Human readable file